Unify feature probing packages

rgo3 opened the pull request:
With this PR we remove the `HaveFullLPM()` feature probe as it is no longer needed since cilium bumped its minimal supported kernel version to 4.19.

We also migrate any leftover APIs from `pkg/probe` to `pkg/datapath/linux/probes` to only have one go `pkg` for feature probing APIs.

Fixes: #25621

rgo3 committed:
probe: remove HaveFullLPM feature probe

Since Cilium's minimal required kernel version was bumped to 4.19 all
supported kernels have full LPM map support. This allows us to remove
this feature probe as it would always be true.

Signed-off-by: Robin Gögge <r.goegge@isovalent.com>

rgo3 committed:
probes: migrate leftover API from pkg/probe

This also refactors HaveIPv6Support into returning an error instead of a
bool to keep feature probing APIs consistent across the package and adds
a small test for the probe.

Fixes: #25621.

Signed-off-by: Robin Gögge <r.goegge@isovalent.com>

ti-mo commented on pkg/datapath/prefilter/prefilter.go:
```diff
@@ -250,13 +243,9 @@ func (p *PreFilter) init() (*PreFilter, error) {
-	if !haveLPM {
-		log.Warning("Kernel too old for full LPM map support. Needs kernel 4.16 or higher. Only enabling /32 and /128 prefixes for prefilter.")
-	}
 	c := preFilterConfig{
```
It looks like `preFilterConfig` can only ever have all its flags set to `true`, so we should probably remove it in a subsequent commit.

ti-mo commented on pkg/datapath/linux/probes/probes.go:
```diff
@@ -504,6 +504,18 @@ func HaveOuterSourceIPSupport() (err error) {
+// HaveIPv6Support tests whether kernel can open an IPv6 socket. This will
+// also implicitly auto-load IPv6 kernel module if available and not yet
+// loaded.
+func HaveIPv6Support() error {
```
Could we add a test that expects this to succeed unconditionally?

ti-mo reviewed (changes requested):
Thanks! We can remove moar code!

rgo3 committed:
prefilter: remove config struct

With the removal of HaveFullLPM every field in the prefilter config
struct will always be true. This allows the removal of all logic checks
based on those fields and the config struct itself.

Signed-off-by: Robin Gögge <r.goegge@isovalent.com>
